test/TestUtils.py

```diff
@@ -185,3 +187,48 @@ def test_expand_paths_vars(monkeypatch):
     monkeypatch.setenv('TEST_PATH', test_path)
     assert utils.expand_paths_vars(['~']) == [os.path.expanduser('~')]
     assert utils.expand_paths_vars(['$TEST_PATH']) == [test_path]
+
+
+def test_get_yaml_files_no_git_verbose(monkeypatch, capsys):
+    options, _ = cli.get_config(['-v'])
+    monkeypatch.setenv('PATH', '')
+    utils.get_yaml_files(options)
+    msg1, msg2 = capsys.readouterr().out.splitlines()[:2]
+    assert msg1 == 'Discovering files to lint: git ls-files *.yaml *.yml'
+    assert msg2.startswith('Warning: Failed to locate command:')
+
+
+def test_get_yaml_files_not_git_verbose(monkeypatch, capsys):
+    options, _ = cli.get_config(['-v'])
+    monkeypatch.chdir('/tmp')
+    utils.get_yaml_files(options)
+    msg1, msg2 = capsys.readouterr().out.splitlines()[:2]
+    assert msg1 == 'Discovering files to lint: git ls-files *.yaml *.yml'
+    assert msg2.startswith('Warning: Failed to discover yaml files to lint using git:')
+
+
+@pytest.mark.parametrize('is_in_git', (True, False))
+def test_get_yaml_files_silent(is_in_git, monkeypatch, capsys, tmp_path):
+    options, _ = cli.get_config([])
+    test_dir = Path(os.path.dirname(os.path.realpath(__file__)))
+    role_sub_dir = Path('roles/test-role')
+    role_dir = test_dir / 'roles/test-role'
+
+    if is_in_git:
+        lint_path = role_dir
+    else:
+        lint_path = tmp_path / role_sub_dir
+        shutil.copytree(role_dir, lint_path)
+
+    yaml_count = (
+        len(list(lint_path.glob('**/*.yml'))) + len(list(lint_path.glob('**/*.yaml')))
+    )
+
+    monkeypatch.chdir(lint_path)
+    files = utils.get_yaml_files(options)
+    stdout = capsys.readouterr().out
+    assert not stdout, 'No output is expected when the verbosity is off'
+    assert len(files) == yaml_count, "Expected to find {count} yaml files in {path}".format(
+        count=yaml_count,
+        path=lint_path
+    )
```

Comment:
There's a nice trick for f-string-like interpolation:
```suggestion
    assert len(files) == yaml_count, "Expected to find {count} yaml files in {path}".format_map(
        locals(),
    )
```